### PR TITLE
Improve build process to support automated beta and stable releases, add new release version, bump dependencies.

### PR DESCRIPTION
**Pull request by belloale**
1. **Github Workflows Improvements**
Resolving some of the pending items with the build process. GitHub Actions will publish a new version to `beta` on every commit to master and releases to `stable` can be manually requested by project owners when new releases are completed.

2. **New Release Version**
Added IntelliJ 2021.1+ support to the plugin and back ported support to Intellij 2020.2+ vs 2020.3+ to pick up Android Studio support. Fixed several of the warnings with Kotlin dependencies by clamping the `apiVersion` to the version supported by that particular IntelliJ platform.

3. **Bump Dependencies**
Bumped Kotlin to `1.5.x` and the IntelliJ Gradle Plugin to `1.1.x`.

**Future Considerations**
* Clamping `apiVersion` may cause issues where the Kotlin Complier may not support restricting the API to previous versions of IntelliJ, therefore we may either need to phase out support for older IntelliJ versions, maybe at a frequency of the last three minor versions of IntelliJ.
* Version `1.1.x` of the IntelliJ Gradle Plugin adds support for a `signPlugin` target which lets us sign the produced `.zip` with our credentials. We should investigate doing this to improve the security of this distro.
* Consider adding `runPluginVerifier` target to the build process as it verifies binary compatibility of the produced artifacts.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -27,55 +29,17 @@ jobs:
...
-  # Build plugin with buildPlugin Gradle task and provide artifact.
+  # Build and test plugin and provide artifact.
   build:
     name: Build
-    needs: test
+    needs: gradleValidation
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        product: [ "IC-2020.2", "IC-2021.1" ]
```
Comment (hiljusti) [inline .github/workflows/build.yml:39]: Awesome!

**Review by hiljusti — APPROVED** [review]
Thanks for doing this!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -128,9 +92,20 @@ jobs:
...
         run: ./gradlew buildPlugin
 
+      # Run intellij:verifyPlugin task.
+      - name: Verify Plugin
+        run: ./gradlew verifyPlugin
+
+      # Publish plugin to beta channel
+      - name: Publish Beta Plugin
+        env:
+          PUBLISH_TOKEN: ${{ secrets.JETBRAINS_TOKEN }}
```
Comment (tgregg) [inline .github/workflows/build.yml:102]: Where does this token live?

**Review by tgregg — APPROVED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (belloale) [reply .github/workflows/build.yml:102]: Can't tell you that. It's a secret.

Ok so real answer, if you have Collaborator access you will see `Settings` -> `Secrets`. Github does not allow you to view the value from the UX and only Github Workflows triggered by people with Collaborator access will be able to access the value from within the workflow.

**Pull request merged**